### PR TITLE
docs: update CONTRIBUTING guide, fix PR template, and clean up README

### PR DESCRIPTION
**Pull request by SahilAujla**
## PR-Codex overview
This PR updates the pull request checklist and adds more detailed instructions in the README and CONTRIBUTING.md files.

### Detailed summary
- Updated the pull request checklist in the PULL_REQUEST_TEMPLATE.md file.
- Added more detailed instructions and information in the README.md file.
- Updated the CONTRIBUTING.md file with a step-by-step guide for contributors.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,24 +1,58 @@
...
-- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples? (e.g. `feat!: breaking change`)
-- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:fix`)
-- [ ] Is the base branch you're merging into `development` and not `main`?
+2. **Install Dependencies**: Ensure you have `yarn` installed (we use version 1.x). Installation instructions are available [here](https://classic.yarnpkg.com/lang/en/docs/install). Run `yarn` in the project root to install all necessary dependencies.
 
-## Docs guidelines
+3. **Testing Environment**: Before making changes, make sure to verify the testing environment.
 
-We leverage `vitepress` to build our docs. You can find the docs in the `site` folder. To run the docs locally, run `yarn docs:dev` from the `site` folder. To build the docs, run `yarn docs:build`. Docs are automatically deployed with each package published.
+   - Use the Node version specified in `package.json` (currently 18.10). Run `node -v` to check your version.
```
Comment (avasisht23) [inline CONTRIBUTING.md:13]:
```suggestion
   - Use the Node version specified in `package.json` (currently 18.16.0). Run `node -v` to check your version.
```

File: README.md [hunk 2]
```diff
@@ -1,50 +1,15 @@
-# Account Abstraction SDK (aa-sdk)
+# aa-sdk
 
-The `aa-sdk` is a type-safe and performant TypeScript library built on top of [viem](https://viem.sh/) to provide ergonomic methods for sending user operations, sponsoring gas, and deploying smart accounts. It handles all the complexity of ERC-4337 under the hood to make account abstraction simple.
+The `aa-sdk` is a type-safe and performant TypeScript library built on top of [viem](https://viem.sh/) to provide ergonomic methods for sending user operations, sponsoring gas, and deploying smart contract accounts. It handles all the complexity of ERC-4337 under the hood to make account abstraction simple.
```
Comment (avasisht23) [inline README.md:3]:
```suggestion
The `aa-sdk` is a type-safe and performant TypeScript library built on top of [viem](https://viem.sh/) to provide ergonomic methods for sending user operations, sponsoring gas, and deploying smart accounts. It handles all the complexity of ERC-4337 under the hood to make account abstraction simple.
```

File: README.md [hunk 3]
```diff
@@ -1,50 +1,15 @@
...
 
 The SDK also implements an EIP-1193 provider interface to easily plug into any popular dapp or wallet connect libraries such as RainbowKit, Wagmi, and Web3Modal. It also includes ethers.js adapters to provide full support for ethers.js apps.
 
-The `aa-sdk` is modular at every layer of the stack and can be easily extended to fit your custom needs. You can plug in any [smart account](https://accountkit.alchemy.com/smart-accounts/accounts/using-your-own) implementation, [Signer](https://accountkit.alchemy.com/smart-accounts/signers/overview), [Gas Manager API](https://accountkit.alchemy.com/overview/introduction.html#gas-manager-api), RPC provider.
+The `aa-sdk` is modular at every layer of the stack and can be easily extended to fit your custom needs. You can plug in any [smart account](https://accountkit.alchemy.com/smart-accounts/accounts/using-your-own) implementation, [Signer](https://accountkit.alchemy.com/smart-accounts/signers/overview), Gas Manager API and RPC Provider.
```
Comment (avasisht23) [inline README.md:7]: hyperlink to gas manager docs?

**Review by avasisht23 — APPROVED** [review]
nits, then good to go

File: README.md (hunk 3 above)
Comment (SahilAujla) [reply README.md]: Done!

**Pull request merged**